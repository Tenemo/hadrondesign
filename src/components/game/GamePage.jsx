import React from 'react';
import { connect } from 'react-redux';
import * as gameActions from '../../actions/gameActions';

export class GamePage extends React.Component {
    constructor(props, context) {
        super(props, context);

        this.state = {
            game: { title: '' }
        };

        this.onTitleChange = this.onTitleChange.bind(this);
        this.onClickSave = this.onClickSave.bind(this);
    }

    onTitleChange(event) {
        const game = this.state.game;
        game.title = event.target.value;
        this.setState({ game: game });
    }

    onClickSave() {
        this.props.dispatch(gameActions.newGame(this.state.course));
        //console.log(`Saving ${this.state.game.title}`);
    }

    render() {
        return pug`
            div
                h1 Flip 'Em
                h2 New Game
                input(
                    type='text'
                    onChange=this.onTitleChange
                    value=this.state.game.title
                )
                input(
                    type='submit'
                    value='Save'
                    onClick=this.onClickSave
                )
        `;
    }
}
function mapStateToProps(state) {
    return {
        games: state.games
    };
}
export default connect(
    mapStateToProps
)(GamePage);
