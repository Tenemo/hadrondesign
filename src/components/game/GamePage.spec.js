import React from 'react';
import GamePage from './GamePage';

describe('App', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(pug`
            GamePage
        `);
        expect(wrapper).toMatchSnapshot();
    });
});