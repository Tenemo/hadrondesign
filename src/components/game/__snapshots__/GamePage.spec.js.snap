// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamePage matches snapshot 1`] = `
<div>
  <h1>
    Flip 'Em
  </h1>
  <h2>
    New Game
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    onClick={[Function]}
    type="submit"
    value="Save"
  />
</div>
`;
